ee/codegen/src/generators/nodes/tool-calling-node.ts

```diff
@@ -0,0 +1,94 @@
+import { python } from "@fern-api/python-ast";
+import { Field } from "@fern-api/python-ast/Field";
+import { AstNode } from "@fern-api/python-ast/core/AstNode";
+
+import { ToolCallingNodeContext } from "src/context/node-context/tool-calling-node";
+import { NodeOutputs } from "src/generators/node-outputs";
+import { NodeTrigger } from "src/generators/node-trigger";
+import { BaseSingleFileNode } from "src/generators/nodes/bases/single-file-base";
+import { WorkflowValueDescriptor } from "src/generators/workflow-value-descriptor";
+import { ToolCallingNode as ToolCallingNodeType } from "src/types/vellum";
+import { toPythonSafeSnakeCase } from "src/utils/casing";
+
+export class ToolCallingNode extends BaseSingleFileNode<
```

Comment:
[nit] similar to above, I'd like to relay on Base Node's default file implementation as much as possible